### PR TITLE
[HttpClient] Add `JsonMockResponse`

### LINKED ISSUE #18230: [HttpClient] Add `JsonMockResponse`
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#50044
| PR author(s) | @fancyweb
| Merged in    | 6.3

We created this issue to not forget to document this new feature. We would really appreciate if you can help us with this task. If you are not sure how to do it, please ask us and we will help you.

To fix this issue, please create a PR against the 6.3 branch in the [symfony-docs repository](https://github.com/symfony/symfony-docs).

Thank you! :smiley:

### PR DESCRIPTION
**Pull request by alamirault**
Fix #18230

### CODE REVIEW

**Review by MrYamous — APPROVED** [review]

File: http_client.rst [hunk 1]
```diff
@@ -2032,6 +2032,19 @@ Then configure Symfony to use your callback:
             ;
         };
 
+You can also use :class:`Symfony\\Component\\HttpClient\\Response\\JsonMockResponse` which automatically encode
```
Comment (OskarStark) [inline http_client.rst]: We should explain before and after, like:

"To return json, you would normally do ......", CODE-EXAMPLE, "you can use the following way instead...." CODE-EXAMPLE with `JsonMockResponse`

WDYT?

Comment (alamirault) [reply http_client.rst]: I push new version with your suggestion !

But I'm not sure about php code style, What is the rule ? Does pipeline check this ?

**Review by OskarStark — APPROVED** [review]

File: http_client.rst [hunk 2]
```diff
@@ -2032,6 +2032,30 @@ Then configure Symfony to use your callback:
             ;
         };
 
+To return json, you would normally do::
+
+    use Symfony\Component\HttpClient\Response\MockResponse;
+
+    $response = new MockResponse(json_encode([
+        'foo' => 'bar',
+    ]), [
```
Comment (OskarStark) [inline http_client.rst]:
```suggestion
            'foo' => 'bar',
        ]), [
```

**Comment by OskarStark** [comment]
Thanks Antoine.

**Pull request merged**